src/v/cluster/rm_stm.cc

```diff
@@ -2295,7 +2295,31 @@ ss::future<stm_snapshot> rm_stm::take_snapshot() {
             tx_ss.seqs.push_back(entry.second.copy());
         }
         tx_ss.offset = _insync_offset;
+
+        for (const auto& entry : _log_state.tx_seqs) {
+            tx_ss.tx_seqs.push_back(tx_snapshot::tx_seqs_snapshot{
+              .pid = entry.first, .tx_seq = entry.second});
+        }
+
+        for (const auto& entry : _log_state.inflight) {
+            tx_ss.inflight.push_back(tx_snapshot::inflight_snapshot{
+              .pid = entry.first, .counter = entry.second});
+        }
+
+        for (const auto& entry : _log_state.expiration) {
+            tx_ss.expiration.push_back(tx_snapshot::expiration_snapshot{
+              .pid = entry.first, .timeout = entry.second.timeout});
+        }
+
         reflection::adl<tx_snapshot>{}.to(tx_ss_buf, std::move(tx_ss));
+    } else if (version == tx_snapshot_v2::version) {
+        tx_snapshot_v2 tx_ss;
+        fill_snapshot_wo_seqs(tx_ss);
+        for (const auto& entry : _log_state.seq_table) {
+            tx_ss.seqs.push_back(entry.second.copy());
+        }
+        tx_ss.offset = _insync_offset;
+        reflection::adl<tx_snapshot_v2>{}.to(tx_ss_buf, std::move(tx_ss));
```

Comment:
why not serde?

Reply:
time constrains, created an issue to follow up after the release - https://github.com/redpanda-data/redpanda/issues/6877